src/projectscene/view/clipsview/clipslistmodel.cpp

```diff
@@ -476,43 +496,86 @@ bool ClipsListModel::trimRightClip(const ClipKey& key, bool completed)
 
 void ClipsListModel::selectClip(const ClipKey& key)
 {
-    selectionController()->setSelectedClip(key.key);
-    selectionController()->setSelectedTracks(TrackIdList({ key.key.trackId }));
+    Qt::KeyboardModifiers modifiers = keyboardModifiers();
+
+    if (modifiers.testFlag(Qt::ControlModifier)) {
+        appendToSelectedClips(key);
+    } else {
+        if (muse::contains(selectionController()->selectedClips(), key.key)) {
+            return;
+        }
+        selectionController()->setSelectedClips(ClipKeyList({ key.key }));
+        selectionController()->setSelectedTracks(TrackIdList({ key.key.trackId }));
+    }
+}
+
+void ClipsListModel::appendToSelectedClips(const ClipKey &key)
+{
+    auto selectedClips = selectionController()->selectedClips();
+    auto selectedTracks = selectionController()->selectedTracks();
+
+    if (!muse::contains(selectedClips, key.key)) {
+        selectedClips.push_back(key.key);
+    }
+    if (!muse::contains(selectedTracks, key.key.trackId)) {
+        selectedTracks.push_back(key.key.trackId);
+    }
+
+    selectionController()->setSelectedClips(selectedClips);
+    selectionController()->setSelectedTracks(selectedTracks);
 }
 
-void ClipsListModel::unselectClip(const ClipKey& key)
+void ClipsListModel::setItemsSelected(bool selected)
 {
-    //! TODO AU4: this will need to be improved when
-    //! having an option to select multiple clips
-    Q_UNUSED(key)
-    selectionController()->resetSelectedClip();
+    for (auto& selectedItem : m_selectedItems) {
+        selectedItem->setSelected(selected);
+    }
 }
 
-void ClipsListModel::resetSelectedClip()
+void ClipsListModel::resetSelectedClips()
 {
-    selectionController()->resetSelectedClip();
+    selectionController()->resetSelectedClips();
+    setItemsSelected(false);
+    m_selectedItems.clear();
 }
 
 void ClipsListModel::onSelectedClip(const trackedit::ClipKey& k)
 {
-    if (m_selectedItem && m_selectedItem->clip().key == k) {
-        return;
+    // ignore if item already selected
+    for (const auto& selectedItem : m_selectedItems) {
+        if (selectedItem->clip().key == k) {
+            return;
+        }
     }
 
-    if (m_selectedItem) {
-        m_selectedItem->setSelected(false);
-    }
+    Qt::KeyboardModifiers modifiers = keyboardModifiers();
 
-    if (m_trackId != k.trackId) {
-        m_selectedItem = nullptr;
+    if (modifiers.testFlag(Qt::ControlModifier)) {
+        if (m_trackId != k.trackId) {
+            return;
+        } else {
+            m_selectedItems.append(itemByKey(k));
+            setItemsSelected(true);
+        }
     } else {
-        m_selectedItem = itemByKey(k);
-        if (m_selectedItem) {
-            m_selectedItem->setSelected(true);
+        setItemsSelected(false);
+
+        if (m_trackId != k.trackId) {
+            m_selectedItems.clear();
+        } else {
+            m_selectedItems = QList<ClipListItem*>({ itemByKey(k) });
+            setItemsSelected(true);
```

Comment:
I suggest slightly improving the implementation of working with selected items.
As far as I can see, all items that are in the `m_selectedItems` list should be "selected".
I suggest adding methods to manage this list and encapsulate this logic there, in one place.
Something like this
```
void ClipsListModel::setSelectedItems(const QList<ClipListItem*>& items)
{
      for (auto& selectedItem : m_selectedItems) {
          selectedItem->setSelected(false);
      }

      m_selectedItems = items;

      for (auto& selectedItem : m_selectedItems) {
          selectedItem->setSelected(true);
      }
}

void ClipsListModel::addSelectedItem(ClipListItem* item)
{
      item->setSelected(true);
      m_selectedItems.append(item);
}

void ClipsListModel::removeSelectedItem(ClipListItem* item)
{
      item->setSelected(false);
      muse::remove(m_selectedItems, item);
}

void ClipsListModel::clearSelectedItems()
{
       for (auto& selectedItem : m_selectedItems) {
          selectedItem->setSelected(false);
      }
      m_selectedItems.clear();
}

```